esiclient/tests/unit/v1/test_node_console.py

```diff
@@ -0,0 +1,102 @@
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+import mock
+
+from osc_lib import exceptions
+
+from esiclient.tests.unit import base
+from esiclient.v1 import node_console
+
+
+class TestNodeConsoleConnect(base.TestCommand):
+
+    def setUp(self):
+        super(TestNodeConsoleConnect, self).setUp()
+        self.cmd = node_console.NodeConsoleConnect(self.app, None)
+
+        self.node_console_1 = {
+            "console_enabled": True,
+            "console_info": {"type": "socat", "url": "tcp://192.168.1.2:8024"}
+            }
+        self.node_console_2 = {
+            "console_enabled": True,
+            "console_info": {"type": "socat", "url": "tcp://192.168.1.2:8025"}
+            }
+        self.node_console_3 = {
+            "console_enabled": False,
+            "console_info": None
+        }
+
+    @mock.patch('esiclient.v1.node_console.os.system',
+                return_value=0, autospec=True)
+    def test_take_action(self, mock_system):
+
+        self.app.client_manager.baremetal.node.get_console.\
+            return_value = self.node_console_1
+
+        arglist = ['node_console_1']
+        verifylist = []
+
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        result = self.cmd.take_action(parsed_args)
+
+        self.assertEqual(0, result)
+        mock_system.assert_called_once
+
+    @mock.patch('esiclient.v1.node_console.os.system',
+                return_value=256, autospec=True)
+    def test_take_action_invalid_tcp_url(self, mock_system):
+
+        self.app.client_manager.baremetal.node.get_console.\
+            return_value = self.node_console_2
+
+        arglist = ['node_console_2']
+        verifylist = []
+
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        result = self.cmd.take_action(parsed_args)
+
+        self.assertNotEqual(0, result)
+        mock_system.assert_called_once
+
+    def test_take_action_no_console_info(self):
+
+        self.app.client_manager.baremetal.node.get_console.\
+            return_value = self.node_console_3
+
+        arglist = ['node_console_3']
+        verifylist = []
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        self.assertRaisesRegex(
+            exceptions.CommandError,
+            'ERROR: No console info for node_console_3. '
+            'Run openstack baremetal node console '
+            'enable for given node',
+            self.cmd.take_action, parsed_args)
+
+    def test_take_action_unknown_console_node(self):
+
+        self.app.client_manager.baremetal.node.get_console.\
+            return_value = None
```

Comment:
Does `get_console` ever actually return `None`? I thought it'd only return a dict or an exception. If so, you can remove this test case and get rid of the if statement I mention below.

Reply:
No. I am removing it.